packages/react-router/lib/hooks.tsx

```diff
@@ -729,6 +740,13 @@ function useCurrentRouteId(hookName: DataRouterStateHook) {
   return thisRoute.route.id;
 }
 
+/**
+ * Returns the ID for the nearest contextual route
+ */
+export function useRouteId() {
+  return useCurrentRouteId(DataRouterStateHook.UseRouteId);
+}
```

Comment:
This doesn't _have_ to be public but I think it's a useful hook for reusable components that may be used across different routes?